src/classes/CCON_CannotDelete_TEST.cls

```diff
@@ -0,0 +1,160 @@
+	CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+	LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+	ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+	POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Course Enrollments
+* @group-content ../../ApexDocContent/CourseEnrollments.htm
+* @description Tests for CCON_CannotDelete_TDTM.
+*/
+@isTest
+private with sharing class CCON_CannotDelete_TEST {
+    @isTest
+    private static void testPreventDeletionCourseConnectionLogic() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Account_Processor__c = UTIL_Describe_API.getHhAccRecTypeID(),
+        																		Prevent_Course_Connection_Deletion__c = true));
+
+        Account acc = UTIL_UnitTestData_TEST.getMultipleTestAccounts(1, UTIL_Describe_API.getBizAccRecTypeID())[0];
+        insert acc;
+
+        Contact[] contacts = UTIL_UnitTestData_TEST.getMultipleTestContacts(6);
+        insert contacts;
+
+        Course_Offering__c courseOffering = UTIL_UnitTestData_TEST.createCourseOffering(NULL, NULL);
+
+        List<Course_Enrollment__c> courseConnections = new List<Course_Enrollment__c>();
+        Course_Enrollment__c courseConnection1 = UTIL_UnitTestdata_TEST.getCourseConnection(contacts[0].Id, courseOffering.Id);
+        Course_Enrollment__c courseConnection2 = UTIL_UnitTestdata_TEST.getCourseConnection(contacts[1].Id, courseOffering.Id);
+        Course_Enrollment__c courseConnection3 = UTIL_UnitTestdata_TEST.getCourseConnection(contacts[2].Id, courseOffering.Id);
+        Course_Enrollment__c courseConnection4 = UTIL_UnitTestdata_TEST.getCourseConnection(contacts[3].Id, courseOffering.Id);
+        Course_Enrollment__c courseConnection5 = UTIL_UnitTestdata_TEST.getCourseConnection(contacts[4].Id, courseOffering.Id);
+        courseConnections.add(courseConnection1);
+        courseConnections.add(courseConnection2);
+        courseConnections.add(courseConnection3);
+        courseConnections.add(courseConnection4);
+        courseConnections.add(courseConnection5);
+        insert courseConnections;
+
+        List<Term_Grade__c> termGrades = new List<Term_Grade__c>();
+        Term_Grade__c termGrade1 = UTIL_UnitTestData_TEST.getTermGrade(contacts[0].Id, courseOffering.Id, courseConnections[0].Id);
+        Term_Grade__c termGrade2 = UTIL_UnitTestData_TEST.getTermGrade(contacts[2].Id, courseOffering.Id, courseConnections[2].Id);
+        termGrades.add(termGrade1);
+        termGrades.add(termGrade2);
+
+        List<Attendance_Event__c> attendanceEvents = new List<Attendance_Event__c>();
+        Attendance_Event__c attendanceEvent1 = UTIL_UnitTestData_TEST.getAttendanceEvent();
+        attendanceEvent1.Contact__c = contacts[1].Id;
+        attendanceEvent1.Course_Connection__c = courseConnections[1].Id;
+        Attendance_Event__c attendanceEvent2 = UTIL_UnitTestData_TEST.getAttendanceEvent();
```

Comment:
Oh looks like we have another getAttendanceEvent() method with 4 parameters can you use that instead? would reduce a few lines of code here.

getAttendanceEvent(Id contactId, Id courseConnectionID, String attendanceType, String reason)

Reply:
@skosgi1 , do you know why we have 2 of the same method one without the signature, and one with?

Reply:
I think the first one was an old one...the second one was created to pass a few paramters I think